op-challenger/game/fault/trace/asterisc/state_converter.go

```diff
@@ -25,6 +26,76 @@ type VMState struct {
 	StateHash common.Hash `json:"stateHash"`
 }
 
+// Because the binary file is provided by Asterisc with all of its fields,
+// we must keep in mind the rest of the fields that aren't used, but still serialized.
+func (s *VMState) Deserialize(in io.Reader) error {
```

Comment:
For cannon we've moved all knowledge of the state format out of op-challenger which makes it possible to support multiple different versions of the state much more easily. We now just run the `cannon witness` subcommand to get it to parse the state and give back the info we need: https://github.com/ethereum-optimism/optimism/blob/368c13326f154d051d9dee9c336c2b7c470e3483/op-challenger/game/fault/trace/cannon/state_converter.go#L35

Would it be possible to do something similar with asterisc? Then we don't need to worry about keeping the challenger implementation of the data format in sync with asterisc and asterisc is free to change the snapshot format in pretty much anyway it wants.  The only catch is that challenger still knows about file extensions so it tries to download `.bin.gz`, `.json.gz` and `.json`.

Reply:
Definitely! I added the usage of `asterisc witness` and it is much simpler as you've mentioned. The logic for running the cmd or distinguishing different file extensions exist both for cannon and asterisc, but I think it's better for the sake of clarity.